lib/json-schema/validator.rb

```diff
@@ -445,13 +448,13 @@ def initialize_schema(schema)
       if schema.is_a?(String)
         begin
           # Build a fake URI for this
-          schema_uri = Addressable::URI.parse(fake_uuid(schema))
+          schema_uri = JSON::Util::URI.parse(fake_uuid(schema))
           schema = JSON::Schema.new(JSON::Validator.parse(schema), schema_uri, @options[:version])
           if @options[:list] && @options[:fragment].nil?
             schema = schema.to_array_schema
           end
           Validator.add_schema(schema)
-        rescue
+        rescue JSON::ParserError
```

Comment:
Currently in master we have [`JSON::Schema::JsonParseError`](https://github.com/ruby-json-schema/json-schema/blob/master/lib/json-schema/validator.rb#L541) being rescued here. But if we only support JSON, I think there is no need to wrap parsing errors any more. Did I miss something?